code-challenges.js

```diff
@@ -18,8 +49,19 @@ var cubeAndSum1 = [2, 3, 4]
 var cubeAndSum2 = [0, 5, 10]
 // Expected output: 1125
 
-
-
+//create a function that takes in an array
+const cubeSum = (array) => {
+    //get an array of all the numbers cubed
+    let cubeArr = array.map(value =>{
+        return Math.pow(value, 3)
+    })
+    //use the reduce() method to add all the numbers in the array together into one sum. 
+    //uses an anonymous function with arguments (a, b) and we are returning the sum of those two numbers. This will continue until there is one number left.
+    //0 is the initial value I give it just in case whatever is entered doesn't provide one to avoid errors
+    return cubeArr.reduce(function(a, b){return a + b}, 0)
+}
```

Comment:
This is AWESOME. Really good work. I'd like you to take a look at some refactoring I did inorder for you to see how we can make this even more concise- 

`const cubeSum = (array) => { return array.map(value =>{ return Math.pow(value, 3) }).reduce(function(a, b){return a + b}, 0)}`

Study this a bit and let me know what you think. Instead of having to declare a new variable to later be returned, since we're already working with a variable we are passing into the function, we might be able to wrap it all up into (I know I say this a lot ;D ) one single return without having to eat up any precious memory we might need to consider in a larger bigger production stage code.

Reply:
Got it! I think in general I should revisit the functions after solving it to see if there's anything I can refactor to make it more concise.